tests/vic-machine.bats

```diff
@@ -12,135 +13,126 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-#!/usr/bin/env bats
-
 load helpers/helpers
 load 'helpers/bats-support/load'
 load 'helpers/bats-assert/load'
 
-installer="${PWD}/../bin/vic-machine"
-docker="$(which docker)"
-log="/sbin"
-
 setup () {
-    assert [ -x "${installer}" ]
+    installer="$(git rev-parse --show-toplevel)/bin/vic-machine"
+    docker="$(which docker)"
+
+    assert [ -n "$VIC_ESX_TEST_URL" ]
+    assert [ -x "$installer" ]
     assert [ -x "$docker" ]
 
-    export GOVC_URL=$VIC_ESX_TEST_URL
-    export GOVC_INSECURE=true
-    export GOVC_PERSIST_SESSION=true
-
-    datacenter=`govc ls / | awk -F/ '{print $2}'`
-    host=`govc ls /${datacenter}/host/`
-    host=`echo ${host} | awk '{print $1}'`
-    POOL_PATH=`govc ls $host/Resources`
-    datastore=`govc ls /$datacenter/datastore | awk -F/ '{print $4}'`
-    IMAGE_STORE_NAME=`echo $datastore | awk '{print $1}'`
-    USER=$(awk -F: '{print $1}' <<<"$GOVC_URL")
-    passurl=$(awk -F: '{print $2}' <<<"$GOVC_URL")
-    PASSWORD=$(awk -F@ '{print $1}' <<<"$passurl")
-    TEST_URL=$(awk -F@ '{print $2}' <<<"$passurl")
-    vch_name="VCH-${RANDOM}"
-    about=`govc about`
-    if [[ "$about" == *"ESX"* ]]; then
-      ISVC=0
-    else
-      ISVC=1
+    export GOVC_URL=$VIC_ESX_TEST_URL GOVC_INSECURE=true
+    export GOVC_USERNAME GOVC_PASSWORD GOVC_RESOURCE_POOL GOVC_DATASTORE
+
+    if [ -z "$GOVC_RESOURCE_POOL" ] ; then
+        GOVC_RESOURCE_POOL=$(govc ls /*/host/*/Resources | head -1)
+    fi
+
+    if [ -z "$GOVC_DATASTORE" ] ; then
+        GOVC_DATASTORE=$(govc ls /*/datastore | head -1 | xargs basename)
     fi
+
+    # split GOVC_URL into GOVC_{URL,USERNAME,PASSWORD}
+    eval "$(govc env | sed -e 's/\$/\\$/g')"
+
+    vch_name="VCH-${RANDOM}"
 }
 
 teardown() {
-    echo "Clear VCH deployments if exists"
-
-    govc vm.destroy "${vch_name}" || echo "no need to delete VCH"
-    govc pool.destroy "${POOL_PATH}/${vch_name}" || echo "no need to delete resource pool"
-    govc datastore.rm -ds ${IMAGE_STORE_NAME} "${vch_name}" || echo "no need to delete datastore files"
-    govc datastore.rm -ds ${IMAGE_STORE_NAME} "VIC" || echo "no need to delete datastore file VIC"
-    govc host.vswitch.remove "${vch_name}" || echo "no need to remove vswitch"
-    rm *.pem || echo "no pem file found"
-    rm *.log || echo "no log file found"
+    if [ -n "$(govc vm.info $vch_name)" ] ; then
+        govc vm.destroy $vch_name
+        govc pool.destroy "$GOVC_RESOURCE_POOL/$vch_name"
```

Comment:
Do you mean to remove the resource pool if VCH VM creation failed?
Cause we don't throw error if VCH resource pool already exists, if VM creation failed, I think we should not remove resource pool as well.
But in test env, it's good to delete resource pool anyway.